### ISSUE TITLE
Incompatability with tc39 decorators proposal

### ISSUE THREAD

**Issue by NilSet** [typeorm/typeorm#2968]
**Issue type:**

[ ] question
[x] bug report
[ ] feature request
[ ] documentation issue

**Database system/driver:**

[ ] `cordova`
[ ] `mongodb`
[ ] `mssql`
[ ] `mysql` / `mariadb`
[ ] `oracle`
[ ] `postgres`
[x] `sqlite`
[ ] `sqljs`
[ ] `react-native`
[ ] `expo`

**TypeORM version:**

[x] `latest`
[ ] `@next`
[ ] `0.x.x` (or put your version here)

**Steps to reproduce or a small repository showing the problem:**
Repro repo: https://github.com/NilSet/typeorm-decorator-incompat

tc39 class decorators are supplied not the class itself, but the descriptor which will be used to create the class.

The export of the module contains the constructed class, so when comparing the export to the annotation capture, they are not `===`. This makes `EntityMetadataBuilder` not find any matching annotated classes.

<img width="1128" alt="screen shot 2018-10-22 at 10 04 44 pm" src="https://user-images.githubusercontent.com/4490501/47336970-a5a66380-d646-11e8-9bb0-23513f12c22f.png">

https://github.com/tc39/proposal-decorators

**Comment by NilSet** [comment]
After hacking together an Entity decorator which works with tc39 decorators, I ran into the same issue with the Column decorators, where it expects a class and instead gets a descriptor.

**Comment by imnotjames** [comment]
We are currently using typescript decorators which are not compatible with the proposal. 

Once the proposal gets through and typescript implements we'll start using that.

**Comment by purerosefallen** [comment]
TypeScript 5.0 comes with the new proposal. Reconsider this?

**Comment by pleerock** [comment]
Yes, we are willing to support tc39 proposal. Feel free to contribute.

**Comment by NSExceptional** [comment]
@pleerock can we reopen this?

**Comment by andrew-demb** [comment]
Hi everyone. 
Does anyone have advice on how to implement support for TC39 decorators in TypeORM?
